q2_diversity_lib/_util.py

```diff
@@ -22,17 +32,63 @@ def _drop_undefined_samples(counts: np.ndarray, sample_ids: np.ndarray,
     return (filtered_counts, filtered_sample_ids)
 
 
-def _disallow_empty_tables(some_function):
-    @wraps(some_function)
-    def wrapper(*args, **kwargs):
-        try:
-            bound_signature = signature(wrapper).bind(*args, **kwargs)
-            table = bound_signature.arguments['table']
-        except KeyError as ex:
-            raise TypeError("The wrapped function has no parameter "
-                            + str(ex) + ".")
-        else:
-            if table.is_empty():
-                raise ValueError("The provided table object is empty")
-        return some_function(*args, **kwargs)
-    return wrapper
+@decorator
+def _disallow_empty_tables(some_function, *args, **kwargs):
+    bound_signature = signature(some_function).bind(*args, **kwargs)
+    table = bound_signature.arguments.get('table')
+    if table is None:
+        raise TypeError("The wrapped function has no parameter 'table'")
+
+    if isinstance(table, BIOMV210Format):
+        table = str(table)
+        table_obj = load_table(table)
+    elif isinstance(table, bTable):
+        table_obj = table
+    else:
+        raise ValueError("Invalid view type: table passed as "
+                         f"{type(table)}")
```

Comment:
I don't have a strong opinion here, but am curious about why you feel this way. Though I agree that a clearly-named variable (e.g. `table_type`) is almost always more readable than a function call, I feel like the ability to inline clear/simple calls in fstrings reduces clutter. There is the slippery slope toward illegible inline function calls to think about, but I'm not sure that's a reason to avoid it entirely. 

As for example 2, I think f strings are simply more readable on a fundamental level. Put the variables where you want them. Experienced developers might be used to c-style string formatting, but that syntax is inherently more complex and harder to read regardless of whether you're using variables or function calls.